pkg/reconciler/broker/broker.go

```diff
@@ -277,25 +270,21 @@ func (r *Reconciler) reconcileUsingCRD(ctx context.Context, b *eventingv1.Broker
 		MarkExchangeFailed(&b.Status, "ExchangeFailure", fmt.Sprintf("Failed to reconcile exchange %q: %s", naming.BrokerExchangeName(args.Broker, false), err))
 		return err
 	}
-	if exchange != nil {
-		if !isReady(exchange.Status.Conditions) {
-			logging.FromContext(ctx).Warnf("Exchange %q is not ready", exchange.Name)
-			MarkExchangeFailed(&b.Status, "ExchangeFailure", fmt.Sprintf("exchange %q is not ready", exchange.Name))
-			return nil
-		}
+	if !exchange.Ready {
+		logging.FromContext(ctx).Warnf("Exchange %q is not ready", exchange.Name)
+		MarkExchangeFailed(&b.Status, "ExchangeFailure", fmt.Sprintf("exchange %q is not ready", exchange.Name))
+		return nil
 	}
 	args.Name = naming.BrokerExchangeName(b, true)
 	dlxExchange, err := r.rabbit.ReconcileExchange(ctx, args)
```

Comment:
it's not clear that dlxExchange is not an Exchange anymore but a Result.

maybe other vars as well

Reply:
It still does represent the exchange, in that the `Ready` and `Name` fields are about the exchange, it's just a different struct than the CRD type. I don't really like calling the variable `result`, since that's just generic. I'm not crazy about `exchangeResult` either, I don't see a problem with it being called `exchange` still.